cmake/SetupTPLs.cmake

```diff
@@ -36,12 +36,23 @@ ENDIF()
 #### AMGX ####################################################################
 IF(${MFMG_ENABLE_CUDA})
   IF (${MFMG_ENABLE_AMGX})
+    FIND_LIBRARY(AMGX_LIBRARY NAMES amgx
+      HINTS ${AMGX_DIR} $ENV{AMGX_DIR}
+      PATH_SUFFIXES lib)
+    FIND_PATH(AMGX_INCLUDE_DIR amgx_c.h
+      HINTS ${AMGX_DIR} $ENV{AMGX_DIR}
+      PATH_SUFFIXES include)
+
+    IF(${AMGX_INCLUDE_DIR} MATCHES "-NOTFOUND" OR ${AMGX_LIBRARY} MATCHES "-NOTFOUND")
+      MESSAGE(FATAL_ERROR 
+              "Error! Could not locate AmgX.")
+    ENDIF()
```

Comment:
Do this instead
```diff
-  IF(${AMGX_INCLUDE_DIR} MATCHES "-NOTFOUND" OR ${AMGX_LIBRARY} MATCHES "-NOTFOUND")
-    MESSAGE(FATAL_ERROR "Error! Could not locate AmgX.")
-  ENDIF()
+  find_package_handle_standard_args(AmgX DEFAULT_MSG AMGX_INCLUDE_DIR AMGX_LIBRARY)
```

Reply:
We found that the macro `find_package_handle_standard_args` would only make sense if we wrote a module to find AmgX and decided to change to `IF(NOT AMGX_INCLUDE_DIR OR NOT AMGX_LIBRARY)`